tests/test_preprocess.py

```diff
@@ -124,3 +132,44 @@ def test_extend_lines(self):
         ext5 = mm.extend_lines(gdf, 2)
         assert ext5.length.sum() > gdf.length.sum()
         assert ext5.length.sum() == pytest.approx(6.2, rel=1e-3)
+
+    def test_roundabout_simplification_1(self):
+        if not GPD_09:
+            with pytest.raises(ImportError):
+                mm.roundabout_simplification(self.df_streets_rabs)
+        else:
+            check = mm.roundabout_simplification(self.df_streets_rabs)
+            assert len(check) == 65
+            assert len(self.df_streets_rabs) == 88  # checking that nothing has changed
+
+    def test_roundabout_simplification_2(self):
+        if GPD_09:
+            check = mm.roundabout_simplification(
+                self.df_streets_rabs, self.df_rab_polys, circom_threshold=0.97
+            )
+            assert len(check) == 77
+            assert len(self.df_streets_rabs) == 88
```

Comment:
```suggestion
    @pytest.mark.skipif(not GPD_09, reason="requires geopandas 0.9+")
    def test_roundabout_simplification_2(self):
        check = mm.roundabout_simplification(
            self.df_streets_rabs, self.df_rab_polys, circom_threshold=0.97
        )
        assert len(check) == 77
        assert len(self.df_streets_rabs) == 88
```

This is another trick pytest can do. Instead of condition you use this decorator and the test is skipped for older geopandas. Use the same for all of them.